compiler/rustc_ty_utils/src/implied_bounds.rs

```diff
@@ -21,14 +21,16 @@ fn assumed_wf_types(tcx: TyCtxt<'_>, def_id: DefId) -> &ty::List<Ty<'_>> {
             assumed_wf_types.extend(liberated_sig.inputs_and_output);
             tcx.intern_type_list(&assumed_wf_types)
         }
-        DefKind::Impl => match tcx.impl_trait_ref(def_id) {
-            Some(trait_ref) => {
-                let types: Vec<_> = trait_ref.substs.types().collect();
-                tcx.intern_type_list(&types)
+        DefKind::Impl => {
+            match tcx.bound_impl_trait_ref(def_id).map(ty::EarlyBinder::subst_identity) {
```

Comment:
this should be `skip_binder` :thinking: ideally `assumed_wf_types` were to also return `EarlyBinder<...>`